luda/editor/client/src/render.ts

```diff
@@ -15,46 +15,93 @@ import {
   SubtitleEditorWithoutSubtitleState,
 } from "./subtitleEditor/type";
 import { Timeline } from "./timeline/Timeline";
-import { TimelineState } from "./timeline/type";
+import { TimelineSequenceNullableState, TimelineState } from "./timeline/type";
 import { saver } from "./saver/saver";
+import { renderSequenceListView } from "./sequenceListView/renderSequenceListView";
+import { SequenceListViewState } from "./sequenceListView/type";
+import { renderTopBar } from "./topBar/renderTopBar";
+import { TopBarState } from "./topBar/type";
 
 type State = {
-  timelineState: TimelineState;
+  timelineState: TimelineSequenceNullableState;
   cameraAngleEditorWithoutCameraAngleState: CameraAngleEditorWithoutCameraAngleState;
   subtitleEditorWithoutSubtitleState: SubtitleEditorWithoutSubtitleState;
   livePlayer: {
     state: LivePlayerState;
     layout: LivePlayerProps["layout"];
   };
+  sequenceListViewState: SequenceListViewState;
+  topBarState: TopBarState;
 };
 
 export function render(state: State): RenderingTree {
-  saver.autoSave("/sequence/sequence1.json", state.timelineState.tracks);
+  const { title, tracks } = state.timelineState;
+  const doesTimelineHasSequence = title && tracks;
+
+  if (!doesTimelineHasSequence) {
+    return renderSequenceListView(
+      {
+        sequenceListView: state.sequenceListViewState,
+        timeline: state.timelineState,
+      },
+      {},
+    );
+  }
+
+  const timelineState: TimelineState = state.timelineState as TimelineState;
```

Comment:
This should be done with type guard because it doesn't guarantee that state.timelineState can be TimelineState.

https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

Reply:
Looks better
fixed with 1a46108